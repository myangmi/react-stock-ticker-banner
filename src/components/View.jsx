import React from 'react';
import PropTypes, { string } from 'prop-types';
import Banner from './Banner.jsx';

import './View.scss';

const propTypes = {
  tickers: PropTypes.arrayOf(string).isRequired,
  apiKey: PropTypes.string,
};

const View = ({ tickers, apiKey }) => (
  <div className='View' >
    <Banner tickers={tickers} apiKey={apiKey} />
  </div>
);

View.propTypes = propTypes;
export default View;
