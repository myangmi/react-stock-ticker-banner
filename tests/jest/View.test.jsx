import React from 'react';
import { shallow } from 'enzyme';

import View from '../../src/components/View.jsx';
import mockTickers from '../mocks/mockTickers';

describe('View', () => {
  it('should render properly', () => {
    const wrapper = shallow(<View tickers={mockTickers} apiKey={'mockAPIKey'}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
