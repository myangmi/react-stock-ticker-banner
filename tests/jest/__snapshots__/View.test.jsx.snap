// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View should render properly 1`] = `
<div
  className="View"
>
  <Banner
    apiKey="mockAPIKey"
    tickers={
      Array [
        "AMD",
        "INTC",
        "TSM",
        "AAPL",
      ]
    }
  />
</div>
`;
